zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/message/DbMessageSubscriptionState.java

```diff
@@ -61,24 +72,30 @@ public DbMessageSubscriptionState(
       final TransactionContext transactionContext,
       final TransientPendingSubscriptionState transientState) {
 
+    subscriptionKey = new DbLong();
     elementInstanceKey = new DbLong();
     messageName = new DbString();
     messageSubscription = new MessageSubscription();
     elementKeyAndMessageName = new DbCompositeKey<>(elementInstanceKey, messageName);
+    elementKeyMessageNameAndSubscriptionKey =
+        new DbCompositeKey<>(elementKeyAndMessageName, subscriptionKey);
     subscriptionColumnFamily =
         zeebeDb.createColumnFamily(
             ZbColumnFamilies.MESSAGE_SUBSCRIPTION_BY_KEY,
             transactionContext,
-            elementKeyAndMessageName,
+            elementKeyMessageNameAndSubscriptionKey,
             messageSubscription);
 
     tenantIdKey = new DbString();
     correlationKey = new DbString();
     tenantAwareNameAndCorrelationKey =
         new DbTenantAwareKey<>(
             tenantIdKey, new DbCompositeKey<>(messageName, correlationKey), PlacementType.PREFIX);
+    tenantAwareNameCorrelationAndElementInstanceKeySubKey =
+        new DbCompositeKey<>(tenantAwareNameAndCorrelationKey, subscriptionKey);
     tenantAwareNameCorrelationAndElementInstanceKey =
-        new DbCompositeKey<>(tenantAwareNameAndCorrelationKey, elementInstanceKey);
+        new DbCompositeKey<>(
+            tenantAwareNameCorrelationAndElementInstanceKeySubKey, elementInstanceKey);
```

Comment:
😟 The same problem I described in the `DbProcessMessageSubscriptionState` exists here. We'll need to migrate this data, adding the `messageSubscriptionKey` but we don't currently have access to this key.

That means we have this problem in 3 column families:
- `PROCESS_SUBSCRIPTION_BY_KEY`
- `MESSAGE_SUBSCRIPTION_BY_KEY`
- `MESSAGE_SUBSCRIPTION_BY_NAME_AND_CORRELATION_KEY`

Oof.